cli/src/pcluster/schemas/cluster_schema.py

```diff
@@ -740,6 +740,12 @@ def no_coexist_role_policies(self, data, **kwargs):
                 "InstanceProfile, InstanceRole or AdditionalIamPolicies can not be configured together."
             )
 
+    @validates_schema
+    def no_coexist_s3_access(self, data, **kwargs):
+        """Validate that instance_role, instance_profile or additional_iam_policies do not co-exist."""
+        if self.fields_coexist(data, ["instance_role", "instance_profile", "s3_access"], **kwargs):
```

Comment:
We already have a validator to check InstanceRole and InstanceProfile co-existence.

So here we can split the checks:
```python
if self.fields_coexist(data, ["instance_role", "s3_access"], **kwargs):
  raise ValidationError("S3Access can not be configured when InstanceRole is set.")
if self.fields_coexist(data, ["instance_profile", "s3_access"], **kwargs):
  raise ValidationError("S3Access can not be configured when InstanceProfile is set.")
```

Reply:
Change to split the checks so we have more clear error message for s3 access validator